std/format/read.d

```diff
@@ -719,3 +719,15 @@ T unformatValue(T, Range, Char)(ref Range input, scope const ref FormatSpec!Char
     int b;
     assertThrown(formattedRead(str, "%s %d-%s", &a, &b, &c));
 }
+
+// https://issues.dlang.org/show_bug.cgi?id=18051
+@safe pure unittest
+{
+    enum Op { lt, gt, eq }
+
+    Op op;
+    Op[] ops;
+    assert(formattedRead!"%s [%(%s, %)]"("lt [lt, gt, eq]", op, ops) == 2);
```

Comment:
Why not the simpler test in the issue description instead?

Reply:
That's a leftover from when I wasn't sure if the same implementation works for arrays and enums (and arrays of enums). I'll change it to the simpler test case.